src/syscheckd/db/src/file.cpp

```diff
@@ -127,7 +130,12 @@ int fim_db_get_not_scanned(fdb_t * fim_sql, fim_tmp_file **file, int storage) {
 
 // LCOV_EXCL_START
 int fim_db_delete_not_scanned(fdb_t * fim_sql, fim_tmp_file *file, pthread_mutex_t *mutex, int storage) {
-    event_data_t evt_data = { .mode = FIM_SCHEDULED, .w_evt = NULL, .report_event = TRUE, .type = FIM_DELETE };
+    event_data_t evt_data;
+    evt_data.report_event = 0;
+    evt_data.mode = FIM_SCHEDULED;
+    evt_data.type = FIM_ADD;
```

Comment:
```suggestion
    evt_data.report_event = TRUE;
    evt_data.mode = FIM_SCHEDULED;
    evt_data.type = FIM_DELETE;
    evt_data.w_evt  = NULL;
```
Deleted events weren't being generated. Aside from this, all other changes looks good to me.

Reply:
Good catch, done in https://github.com/wazuh/wazuh/pull/9225/commits/73fb1b51c19f2e1142293af4ba9ef3f2d4ccad90

I have also rebased the base branch and this branch too